### DISCUSSION TITLE
how to bind device to another router

### DISCUSSION THREAD

**Discussion by avigrinberg** [Koenkk/zigbee2mqtt#19146] [General]
Hi,

i've few devices that uses as router and i want few device to change there routing not to connect directly to the coordinator.
i tried to remove them from the Z2M press the reset button on the device but all the time it choose the coordinator even if it's far away while the device have few meters from him a device that act as router.
i try the bind in the Z2M but the bind is grayed when i choose another router for it. why? how can i force the device to connect to a router and not to coordinator? (i also tried the permit join with specific router but the device isn't connected at all)
PS - all my devices are Tuya zigbee 3.0

![image](https://github.com/Koenkk/zigbee2mqtt/assets/74825241/22faf5fb-2358-44a5-8ad1-5850601f1786)

THX

**Comment by francisp2** [comment]
Bind is not the way to do that. Reset your device, and while pairing only allow pairing through the router you want the device to use.

**Reply by avigrinberg** [reply]
i wrote that i already tried it and in the end it goes to the coordinator according to the  Z2M map
as i wrote i use all Tuya devices with SONOFF ZigBee 3.0 USB Dongle Plus (from factory not flashed). do i need to flush to different firmware?
![image](https://github.com/Koenkk/zigbee2mqtt/assets/74825241/064d09c1-a3de-4081-9d64-2d907e85fb1e)

**Reply by avigrinberg** [reply]
no one can help me here? i did firmware update to latest available and i needed to rejoin all my devices and still every device that is route device connect directly to the coordinator and devices that are EndDevice do connected to route device. why is that? what i'm missing here? please help..

**Comment by francisp2** [comment]
As said, there in the right upper corner, shooce the router you want to pair through

![afbeelding](https://github.com/Koenkk/zigbee2mqtt/assets/52328611/66aaecab-01ff-4830-9cbb-493967c22699)

**Reply by avigrinberg** [reply]
i did that. for some devices that are router devices they are paired good so to speak but after that i go to the z2m map and it show that they connected directly to the coordinator and for some other router devices they not paired at all until i chose the ALL..

**Reply by francisp2** [reply]
Then there is nothing you can do. Zigbee 3 devices should find the best route themselves.

**Reply by sspurgat** [reply]
@avigrinberg did you figure this out? I'm having the same problem white trying to bind Aqara vibration sensors to the Coordinator, because they don't work through routers.

**Comment by tenkaibr** [comment]
I'm having a similar problem. Some switches have duplicate routes and this is causing a problem. I tried to make it connect directly to the coordinator but even when selecting it specifically, it connects to another device. How to force the device to connect to the coordinator?
